src/logging.rs

```diff
@@ -82,16 +83,29 @@ fn add_field_and_payload(data: &mut String, field: &str, payload: &str) {
         data.push_str(&field_payload);
     }
 }
-
-/// Print a message to the journal with the given priority.
+/// Send a message with structured properties to the journal.
 ///
-/// This uses the default systemd-journald socket.
-pub fn journal_print(priority: Priority, msg: &str) -> Result<()> {
+/// vars is a map of journald fields to values.This function uses the default systemd-journald socket.
+/// Iterator may contains `PRIORITY` or `MESSAGE` fields. Instead of duplicating those fields,we'll use the ones which are already injected through journal.send(). So there's a condition in the iterator to skip those two values if present in the map.
```

Comment:
Minor: missing spaces after some of the periods and commas.

Reply:
Re. `PRIORITY` and `MESSAGE`, is that the same behaviour of the `sd_journal_send` API? If so, why not just rely on that directly? If not, why have a different behaviour here?

Reply:
`sd_journal_send` takes a varargs of C format strings plus arguments. This library does not link against `libsystemd.so`, so we are free to shape the API as we wish.

I personally like this API where `print` and `send` are similar, and we just take care of the two special arguments we have for priority and message.
I guess one alternative is to take only the iterator, which makes it more similar to `sd_journal_send`.

Reply:
> This library does not link against `libsystemd.so`

Ahh wow, totally missed that part. Neat!

> I personally like this API where print and send are similar, and we just take care of the two special arguments we have for priority and message.

I think that's OK, but we should be more explicit in the docs that the arguments always override the equivalent fields. This is an ambiguity that the API avoids entirely.

Maybe rewording as something like:

> /// The `PRIORITY` or `MESSAGE` fields from the `vars` iterator are always ignored in favour of the `priority` and `message` arguments.

?